docs/concepts/function-components.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Function Components
+sidebar_label: Introduction
+description: Introduction to function components 
+---
+
+:::warning
+We're still working on function components and hooks. They're not quite ready to be used yet.
+If you'd like to help out, take a look at the [project board](https://github.com/yewstack/yew/projects/3) for a list of things that still need to be done.
+:::
+
+
+Function components are a simplified version of normal components.
+They consist of a single function that receives props and determines what should be rendered by returning `Html`.
+Basically, it's a component that's been reduced to just the `view` method.
+On its own that would be quite limiting because you can only create pure components, but that's where hooks come in.
+Hooks allow function components to use state and other Yew features without implementing the `Component` trait.
+
+## Example:
+ 
+```rust
+#[function_component(HelloWorld)]
+fn hello_world() -> Html {
+    html! {
+        { "Hello world" }
+    }
```

Comment:
We can simplify this down to this:
```suggestion
    html! { "Hello world" }
```

Reply:
It's fine to leave it as is if you prefer it that way, but I was thinking it could look something like this:
```rust
## Creating function components

The easiest way to create a function component is to add the [`#[function_component]`](function-components/attribute.md) attribute to a function.

```rust
#[function_component(HelloWorld)]
fn hello_world() -> Html {
    html! { "Hello world" }
}
<CODE BLOCK END cuz GitHub bad>

```

I don't really think of this as an "example", more of a "visual aid" to give some context to the previous sentence. It makes it clear what we mean by "attribute" (some users might not be familiar with them), and already establishes some familiarity with the required signature.

Reply:
That is a valid point, especially since there's an example on the other page